### PR TITLE
CPU-Adam: add compile-flag to enable param-copy from CPU to GPU

### PR DESCRIPTION
**Pull request by RezaYazdaniAminabadi**
This addresses https://github.com/microsoft/DeepSpeed/issues/1279

### CODE REVIEW

File: csrc/adam/cpu_adam.cpp [hunk 1]
```diff
@@ -102,11 +103,16 @@ void Adam_Optimizer::Step(float* _params,
             SIMD_STORE(_exp_avg + i, momentum_4.data);
             SIMD_STORE(_exp_avg_sq + i, variance_4.data);
         }
+#if defined(__ENABLE_CUDA__)
```
Comment (tjruwase) [inline csrc/adam/cpu_adam.cpp:106]: I believe all the cuda and cublas header files should be guarded as well.  We would probably need to test the build in a CPU-only environment.

Comment (RezaYazdaniAminabadi) [reply csrc/adam/cpu_adam.cpp:106]: I agree, I will add more guards for them too.

File: op_builder/builder.py [hunk 2]
```diff
@@ -267,6 +273,12 @@ def cpu_arch(self):
             return '-mcpu=native'
         return '-march=native'
 
+    def is_cuda_available(self):
+        if torch.cuda.is_available():
```
Comment (tjruwase) [inline op_builder/builder.py:277]: Will this not throw an import exception if torch.cuda is not available? Should this be in a try/except?

Comment (RezaYazdaniAminabadi) [reply op_builder/builder.py:277]: good point 👍 I will add that. Thanks

**Comment by molly-smith** [comment]
Closing since https://github.com/microsoft/DeepSpeed/pull/2507 should address this.

**Pull request closed without merging**